draft-ietf-scitt-architecture.md

```diff
@@ -388,32 +388,36 @@ Registration Policies refer to additional checks over and above the Mandatory Re
 
 #### Mandatory Registration Checks
 
-Transparency Services MUST, at a minimum, perform the following checks before registering a Signed Statement:
+During registration, a Transparency Services MUST, at a minimum, authenticate the Issuer of the Signed Statement by validating the COSE signature and checking the identity of the issuer against one of its configured trust anchors, using the `x5t`, `x5chain` and `kid` protected headers of the Signed Statement as hints.
+For instance, for X.509 signed claims, the Transparency Service must build and validate a complete certificate chain from the Issuer's certificate identified by `x5t`, to one of the root certificates most recently registered as a trust anchor of the Transparency Service.
+
+The Transparency Service MUST evaluate the registration policy that was most recently added to the Append-only Log for instance-specific registration checks.
 
-- Authenticate the Issuer of the Signed Statement
+#### Auditability of Registration
 
-The Transparency Service MUST authenticate the Issuer of Signed Statements by validating the COSE signature and checking the identity of the issuer through one of its configured trust anchors, using the `x5t` and `kid` headers in the protected header as hints. For instance, for X.509 signed claims the Transparency Service must validate a complete certificate chain from the certificate identified by `x5t` to one of the trusted root authority certificate of the Transparency Service.
-The public key is used to verify digital signatures, and the associated data is used to constrain the types of information for which the trust anchor is authoritative."
+The operator of a Transparency Service MAY update the registration policy or the trust anchors of a transparency service at any time.
+This presents a challenge to Auditors and Relying Parties if they cannot determine what policy was used to register a given Transparent Statement.
 
+Transparency Services MUST ensure that for any Transparent Statment they accept, enough information is made available to Auditors (either in the Append-only Log and retrievable through audit APIs, or included in the Receipt) to identify and recover the Transparent Statements describing the registration policy and trust anchors that were in use at the time that Transparent Statement was registered.
 
-Before a Signed Statement is registered, the trust anchor used to verify its Issuer MUST be registered with the Transparency Service.
+In particular, this information SHOULD enable Auditors to re-validate the mandatory Issuer authentication check, unless the required information isn't available (e.g. the authentication protocol is not verifiable), or it is confidential (e.g. an OAuth access token embedding the Issuer's pubic key), or it is private (e.g. a Issuer certificate containing the name and email address of a developer).
```

Comment:
I agreee it is phrased ambiguously, trying to avoid being too prescriptive on implementers and giving them several possible opt-outs. If you think you can do better please provide an alternative

Reply:
I could live with this I think.

```suggestion
The trust anchor used to verify its Issuer MUST be registered with the Transparency Service in order that it is available to Auditors. It MAY be registered at any time.
```

Reply:
I don't this we should write MUST statements regarding Transparency Service policies like this.

I think policies should be out of scope.

Reply:
Would a revoked "trust anchor" certificate have any effect on the statements contained on the chain? Would all of the statement still remain valid in the event of a revoked certificate?

Reply:
Well I know there has been chatter about encoding a hash of the cert chain versus the whole thing and now this question posed by Dick. This leads to a problem, what if the certificates no longer exist, are not just recorded but presently invalid?

Reply:
Spot on AJ. Another problem we see is the inability to verify a certificate using OSCP. Things go wrong. What is the "failsafe" position in the "state machine" for the Transparency Service when things go wrong?

Reply:
SHOULD works for me. I only used MUST because I thought there was an appetite for normative requirements here

Reply:
It is completely OK if the certificate used to sign a statement is revoked or expires after the statement is registered. The point of the "SHOULD" is that the evidence to prove that the certificate was valid (e.g. the intermediate CAs, the valid OCSP certificate with a timestamp around registration time) should be available to Auditors so they can retrospectively verify that the signature was valid at registration.